client_test.go

```diff
@@ -0,0 +1,57 @@
+package osrm
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_getWithError(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(500)
+		fmt.Fprint(w, "something wrong")
+	}))
+	defer ts.Close()
+
+	c := newClient("/", ts.Client())
```

Comment:
As I understand, in this scenario created server is not called at all - right? So it could be smth like `c := newClient("invalid-url", http.Client{})` here? BTW usage of `"/"` is a bit confusing for me (since that's supposed to be some base URL - not relative path or smth)